var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -51,14 +51,21 @@ class Dealii(Package):
     variant('petsc',    default=True,  description='Compile with Petsc (only with MPI)')
     variant('slepc',    default=True,  description='Compile with Slepc (only with Petsc and MPI)')
     variant('trilinos', default=True,  description='Compile with Trilinos (only with MPI)')
+    variant('python',   default=True,  description='Compile with Python bindings')
 
     # required dependencies, light version
     depends_on("blas")
     # Boost 1.58 is blacklisted, see
     # https://github.com/dealii/dealii/issues/1591
     # Require at least 1.59
-    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",     when='~mpi')  # NOQA: ignore=E501
-    depends_on("boost@1.59.0:+mpi+thread+system+serialization+iostreams", when='+mpi')  # NOQA: ignore=E501
+    # +python won't affect @:8.4.1
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",            when='@:8.4.1~mpi')         # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",        when='@:8.4.1+mpi')         # NOQA: ignore=E501
+    # since @8.5.0: (and @develop) python bindings are introduced:
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams",            when='@8.5.0:~mpi~python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi",        when='@8.5.0:+mpi~python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+python",     when='@8.5.0:~mpi+python')  # NOQA: ignore=E501
+    depends_on("boost@1.59.0:+thread+system+serialization+iostreams+mpi+python", when='@8.5.0:+mpi+python')  # NOQA: ignore=E501
```

Comment:
You can remove all of those NOQA's. I added `depends_on` to the list of exceptions for line length checks in https://github.com/LLNL/spack/commit/ea80c8d3c3a0690dcb357840cfb0db824f44fdc9